### ISSUE TITLE
Shared pool not working when cis is deployed in NodePort mode

### ISSUE THREAD

**Issue by sebidude** [F5Networks/k8s-bigip-ctlr#2492]
@vklohiya we ran it in the debugger and noticed, that the fix which was merged with #2396 does not handle duplicate pool members when the controller is deployed in NodePort mode.

we just added 
```golang
} else { // Controller is in NodePort mode.
    if service.Spec.Type == v1.ServiceTypeNodePort {
        for _, port := range service.Spec.Ports {
            endpointMembers := appMgr.getEndpointsForNodePort(port.NodePort, port.Port)
            for _, newMember := range endpointMembers {
                if _, ok := uniqueMembersMap[newMember]; !ok {
                        uniqueMembersMap[newMember] = struct{}{}
                        members = append(members, newMember)
                }
            }
        }
    }
}
```
and made it working.

Please verify and release a bugfix as soon as possible.

Cheers
Sebi

_Originally posted by @sebidude in https://github.com/F5Networks/k8s-bigip-ctlr/issues/2151#issuecomment-1177401421_

**Comment by sebidude** [comment]
Opened this issue as requested here https://github.com/F5Networks/k8s-bigip-ctlr/issues/2151#issuecomment-1177501284

**Comment by sebidude** [comment]
Hey there,
any chance to pick this up? This is an issue for us since Dec 16, 2021 and the fix is really easy.

**Comment by sebidude** [comment]
@vklohiya @trinaths

**Comment by trinaths** [comment]
@sebidude - ETA 2.10.1 Work in progress

**Comment by mdditt2000** [comment]
@sebidude we released CIS 2.10.1 today. Please can you reach out me at automation_toolchain_pm@f5.com so i can prioritize other fixes/enhancements and know who you are. Thanks CIS PM
